overlay.py

```diff
@@ -171,8 +169,16 @@ def on_connect(self, client, userdata, flags, rc):
 			(e.g. drawing self.base_canvas) """
 		pass
 
-	@abstractmethod
 	def update_data_layer(self):
+		""" Catches any errors that occurs while the data layer is being
+			updated. """
```

Comment:
I've realised that this documentation (and similar descriptions in `backend.py` don't fully describe what happens when you call this function. When you call `update_data_layer`, while yes it does catch any errors that occur, its actual purpose is still to update the data layer. For these comments I think you should keep what you already have, but beforehand keep a description of what actually happens when you call the function. You could copy/paste a lot of this from the original function's comments.